### PR TITLE
bump containerd version

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
bump containerd version to 1.7.3.

### Motivation
Fixes CVE https://github.com/DataDog/datadog-agent/issues/18728

### Additional Notes

https://github.com/DataDog/datadog-agent/pull/16771 needs to be merged first

### Possible Drawbacks / Trade-offs
N/A

### Describe how to test/QA your changes
Deploy the Agent on a kubernetes cluster. Make sure containerd events are still received and processed properly: `agent workload-list -v` should show new containers.

We should make sure that the container check still works when using "containerd" as a collector (not "system"). We should test that datadog events are still emitted for containerd (mentioning it just in case, because I'm not sure if you were referring to this or only to workloadmeta events).

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/workloadmeta/collectors/internal/containerd/event_builder.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/containerd/containerd"
 	"github.com/containerd/containerd/api/events"
 	containerdevents "github.com/containerd/containerd/events"
-	"github.com/gogo/protobuf/proto"
+	"google.golang.org/protobuf/proto" // can't use gogo proto until https://github.com/gogo/protobuf/issues/678 is fixed
```
Comment (clamoriniere) [inline pkg/workloadmeta/collectors/internal/containerd/event_builder.go:17]: do we need this comment? It seems that it is not plan to use back to gogo proto.

**Review by clamoriniere — COMMENTED** [review]
looks great 👍

File: pkg/workloadmeta/collectors/internal/containerd/event_builder.go (hunk 1 above)
Comment (AliDatadog) [reply pkg/workloadmeta/collectors/internal/containerd/event_builder.go]: Indeed, I removed it

**Review by clamoriniere — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -257,7 +257,7 @@ require (
 	code.cloudfoundry.org/diego-logging-client v0.0.0-20200130234554-60ef08820a45 // indirect
 	code.cloudfoundry.org/executor v0.0.0-20200218194701-024d0bdd52d4 // indirect
 	code.cloudfoundry.org/go-diodes v0.0.0-20190809170250-f77fb823c7ee // indirect
-	code.cloudfoundry.org/go-loggregator v7.4.0+incompatible // indirect
+	code.cloudfoundry.org/go-loggregator v7.4.0+incompatible // indirect; indirsect
```
Comment (davidor) [inline go.mod:260]: indirsect?

Comment (AliDatadog) [reply go.mod:260]: Indeed surprising. I'll fix that

**Review by chouquette — APPROVED** [review]
LGTM for agent-platform files

**Review by davidor — APPROVED** [review]

**Pull request merged**